### ISSUE TITLE
Plugin state cached by gradle daemon between runs?

### ISSUE THREAD

**Issue by GrafDiffusore** [melix/jmh-gradle-plugin#114]
Not sure if this is daemons fault or plugin (or mine), but whenever I'm adding new `@Benchmark`-annotated methods under `src/jmh` or remove some, I need to do `gradle --stop` for plugin to detect changes. Otherwise it either doesn't run newly added methods, or complains about removed methods:
```
$ ./gradlew :shared:clean :shared:jmh
...
.../actors0/shared/build/jmh-generated-sources/vic/shared/utils/colls/generated/TestToMaps_testBetterScalaToMap_jmhTest.java:320: error: cannot find symbol
            blackhole.consume(l_testtomaps0_0.testBetterScalaToMap());
                                             ^
  symbol:   method testBetterScalaToMap()
  location: variable l_testtomaps0_0 of type TestToMaps_jmhType
```

I'm using plugin 0.4.5 with gradle 4.3.1 (on Debian). Using 0.4.4 with gradle 4.1 seems to be affected in the same way.

**Comment by momaza** [comment]
I got the same issue just this morning and resolved it the same way. If you change the name of a benchmark the task fails unless you stop the daemon first (`gradle --stop`). BTW, building with no daemon (`gradle --no-daemon jmh`) also fails if you have already run with daemon without stopping it.

**Comment by tinca** [comment]
I am also experiencing this and resolve it same way.

**Comment by blacklion** [comment]
It is closed, but I have same problem with 5.4.1 on gradle 

It pick up new benchmark files, but fail to notice that old one were changed even if I celanup `build` directory. Tons of wired errors — not found methods, wrong parameters (if I change `@Param`-annotated fileds), etc.

**Comment by tomwhoiscontrary** [comment]
Same with plugin 0.4.8 and Gradle 5.3.

AFACIT there is no way to disable the daemon in the build.gradle, so i just have to remember to add `--stop` to the command line every time.

**Comment by fedotxxl** [comment]
Why is it closed? This is a very confusing bug. Please fix it or fix documentation

**Comment by tomwhoiscontrary** [comment]
I believe that @melix closed this as a [duplicate of #114](https://github.com/melix/jmh-gradle-plugin/issues/132#issuecomment-404792929).

**Comment by sudhanshugarg** [comment]
Still happening for me, and neither gradle --stop nor gradle --no-daemon seem to work for me. It doesn't recognize new @Benchmark methods, and neither does it recognize additional values added to @Param notation.
